sdk/src/trace/always_on_sampler.cc

```diff
@@ -0,0 +1,34 @@
+#include "opentelemetry/sdk/trace/always_on_sampler.h"
+
+#include "opentelemetry/sdk/common/atomic_shared_ptr.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace trace
+{
+/**
+ * Return RECORD_AND_SAMPLE and an empty set of attributes
+ */
+SamplingResult AlwaysOnSampler::ShouldSample(
+    const SpanContext *parent_context,
+    trace_api::TraceId trace_id,
+    nostd::string_view name,
+    trace_api::SpanKind span_kind,
+    const nostd::span<AttributeKeyValue> &attributes) noexcept
+{
+  return
+  {
+    sdk::trace::Decision::RECORD_AND_SAMPLE,
+        std::unique_ptr<nostd::span<AttributeKeyValue>>(new nostd::span<AttributeKeyValue>())
+  };
```

Comment:
This method of SamplingResult construction isn't ideal in my opinion. Especially once we start creating Samplers with Attributes (i.e. ProbabilitySampler), it's going to get even unclear. I think we should look at creating a class with helper functions similar to how it's implemented in Java.

Reply:
IMO there are three options:
1. return `unique_ptr<SamplingResult>` where SamplingResult is a class.
2. return `SamplingResult` where `SamplingResult` is a struct.
The first option is not ideal when we want to reduce heap allocation.
The second  is that the `SamplingResult` has same functionality for all built-in samplers (See our discussion here: https://github.com/open-telemetry/opentelemetry-cpp/pull/118#discussion_r442439171).  
In Java, they created a `DecisionImpl` that basically works as a builder for `Decision`. And `AlwaysOn` and `AlwaysOff` sampler return a `Decision` without attribute. We can achieve the same functionality without implementing additional helper functions.

Reply:
Ok, that makes sense for the static samplers (always on/off) - but how do you plan on handling the probabilistic and rate limiting ones, as well as any future samplers that require attributes?

Reply:
Then we can manually add attributes to `SamplingResult`. For example, when we use probabilistic sampler, we can simply add ("sampling.probability", probability_value) to the attribute.